pokemongo_bot/cell_workers/utils.py

```diff
@@ -144,3 +145,57 @@ def print_yellow(message):
 
 def print_red(message):
     print(u'\033[91m' + message.decode('utf-8') + '\033[0m')
+
+def nickname_pokemon(bot,pokemon):
+    """This requies a pokemon object containing all the standard fields: id, ivs, cp, etc"""
+    template = bot.config.nickname_template.lower()
+    if template == "{name}":
```

Comment:
Except for the case where someone's renaming their entire inventory (which won't be common, I imagine), the pokemon can't have been nicknamed already. In the case where someone wants to "force" a revert to original name, they could use a hack (since I don't trim whitespace) `{name} `. That said, I can refactor all this if there's a desire for it.

To answer your other question, yes, we can know a pokemon is nicknamed by the presence of a `nickname` key in the `pokemon_data` dictionary.